napari/components/add_layers_mixin.py

```diff
@@ -917,6 +917,59 @@ def _add_layer_from_data(
 
         return layer
 
+    def save_layers(
```

Comment:
I like making this function able to accept a list of layers or list of names - I wouldn't promote `write_layers_with_plugins`.

I'm not sure about the `Layer.save(path)` method. I see how it is convenient, but it would mean that the `Layer` itself would now need to know all about our plugin machinery which seems like mixing things across levels that shouldn't be mixed - unless you can think of a clean way to implement this

Reply:
Ah - just saw it implemented below - ok, I'm open